### PR TITLE
Fix type of the t function as the argument of getDataImportCronFilter

### PR DESCRIPTION
**Pull request by hstastna**
## 📝 Description

Fix the incorrect type of the `t` function, used as the argument of `getDataImportCronFilter`,
also fix some antipattern in `getDataImportCronFilter`.

## 🎥 Demo
**Before:**
t_before
**After:**
t_after

### CODE REVIEW

**Comment by hstastna** [comment]
@avivtur @metalice @pcbailey @yzamir @vojtechszocs please review

File: src/views/datasources/list/DataSourcesListFilters.ts [hunk 1]
```diff
@@ -5,12 +5,14 @@ import { RowFilter } from '@openshift-console/dynamic-plugin-sdk';
 
 import { getDataSourceCronJob } from '../utils';
 
-export const getDataImportCronFilter = (t: TFunction): RowFilter<V1beta1DataSource>[] => [
+export const getDataImportCronFilter = (
+  t: TFunction<'plugin__kubevirt-plugin', undefined>,
```
Comment (vojtechszocs) [inline src/views/datasources/list/DataSourcesListFilters.ts:9]: Maybe it's worth adding
```ts
export type KubevirtTFunction = TFunction<'plugin__kubevirt-plugin', undefined>;
```
(or omit the second type argument, unless it's strictly required)

to `src/utils/hooks/useKubevirtTranslation.ts` and update the code to use `KubevirtTFunction` consistently throughout the codebase.

**Comment by metalice** [comment]
Please check if we can remove t function completely and consume it directly in this file. There are example in our code base.

File: src/views/datasources/list/DataSourcesListFilters.ts (hunk 1 above)
Comment (hstastna) [reply src/views/datasources/list/DataSourcesListFilters.ts:9]: That sounds great to me! I'd like do this in a followup PR.

**Comment by hstastna** [comment]
> Please check if we can remove t function completely and consume it directly in this file. There are example in our code base.

i think it is easily, but we would have to rename `getDataImportCronFilter`  this to `use..` and to make a hook from it. Do you agree on that, @metalice ?

**Comment by metalice** [comment]
> > Please check if we can remove t function completely and consume it directly in this file. There are example in our code base.
> 
> I think it is easily doable, but we would have to rename `getDataImportCronFilter` to `use...` and to make a hook from it. Do you agree on that, @metalice ?

you can simply import {t} from "......useKubevirtTranslation" , no need to custom hook, we exported solo t for cases like this when its a util file and there is a function that needs t. so t will not be transferred to the function, but t will be imported directly where needed, you can have a look inside useKubevirtTranslation file

**Comment by hstastna** [comment]
> > > Please check if we can remove t function completely and consume it directly in this file. There are example in our code base.
> > 
> > 
> > I think it is easily doable, but we would have to rename `getDataImportCronFilter` to `use...` and to make a hook from it. Do you agree on that, @metalice ?
> 
> you can simply import {t} from "......useKubevirtTranslation" , no need to custom hook, we exported solo t for cases like this when its a util file and there is a function that needs t. so t will not be transferred to the function, but t will be imported directly where needed, you can have a look inside useKubevirtTranslation file

Okay, so I'll move the `t` function inside the _getDataImportCronFilter_. Thanks! 👍🏽

**Comment by hstastna** [comment]
> > > Please check if we can remove t function completely and consume it directly in this file. There are example in our code base.
> > 
> > 
> > I think it is easily doable, but we would have to rename `getDataImportCronFilter` to `use...` and to make a hook from it. Do you agree on that, @metalice ?
> 
> you can simply import {t} from "......useKubevirtTranslation" , no need to custom hook, we exported solo t for cases like this when its a util file and there is a function that needs t. so t will not be transferred to the function, but t will be imported directly where needed, you can have a look inside useKubevirtTranslation file

@metalice so I've made a change and this has happened:
uncheck_and_fix_not_working_clusters_ maybe
This is what I was talking about when talking about renaming `getDataImportCronFilter` to use the `use` prefix and to make it a hook from it...

**Comment by metalice** [comment]
> > > > Please check if we can remove t function completely and consume it directly in this file. There are example in our code base.
> > > 
> > > 
> > > I think it is easily doable, but we would have to rename `getDataImportCronFilter` to `use...` and to make a hook from it. Do you agree on that, @metalice ?
> > 
> > 
> > you can simply import {t} from "......useKubevirtTranslation" , no need to custom hook, we exported solo t for cases like this when its a util file and there is a function that needs t. so t will not be transferred to the function, but t will be imported directly where needed, you can have a look inside useKubevirtTranslation file
> 
> @metalice so I've made a change and this has happened: use This is what I was talking about when talking about renaming `getDataImportCronFilter` to use the `use` prefix and to make it a hook from it...

that's because u tried to import the hook and not the exported t function, try like this:
image

**Comment by hstastna** [comment]
> that's because u tried to import the hook and not the exported t function, try like this: image

I did it because `import { t } `didn't work for me, not sure why, I'm still investigating..

**Pull request merged**